src/datagrid/DataGrid.tsx

```diff
@@ -583,6 +638,20 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
         }
     };
 
+    // Function to handle CustomPageSize change in input box
+    private handleCustomPageSizeChange = () => {
+        const {maxCustomPageSize, currentPage} = this.state.pagination!;
+        const pageSize = this.customPageSizeRef.current && parseInt(this.customPageSizeRef.current.value);
+
+        if (pageSize && maxCustomPageSize && pageSize <= maxCustomPageSize) {
+            this.getPage(this.state.pagination!.currentPage, pageSize);
+        } else if (pageSize && maxCustomPageSize && pageSize > maxCustomPageSize) {
+            if (this.customPageSizeRef.current) {
+                this.customPageSizeRef.current.value = maxCustomPageSize.toString();
+                this.getPage(currentPage, parseInt(this.customPageSizeRef.current.value));
```

Comment:
We can not set a user to the current page with max custom page size. 
- Suppose the user has 500 records
- If use is of page 2 with page size as 100
- User enters 2000 as the custom page size
- Since 2000 > 1000 (max custom page size) in the custom textbox user will see 1000
- But it is still on 2nd page which does not exist

I think we should always move user to first page here

Reply:
yes Arpita, thanks for pointing this out